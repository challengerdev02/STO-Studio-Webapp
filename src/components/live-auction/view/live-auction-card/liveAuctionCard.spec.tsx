import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LiveAuctionCard } from './index';

describe('LiveAuctionCard component', function () {
  afterEach(cleanup);
  let props = {
    GENRE_OPTIONS: 'Science Fiction',
    language: 'Japanese',
    ageRating: '13 and above',
    characters: ['Spiderman', 'Magnito'],
    imageSrc: '/live-auction-card-cover.svg',
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<LiveAuctionCard {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('should test the LiveAuctionCard component displays ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('live-auction-card')).toBeInTheDocument();
  });
  it('should test the LiveAuctionCard component has a display image ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('display-image')).toBeInTheDocument();
  });

  it('should test the source attribute of LiveAuctionCard component display image has a value ', () => {
    const { getByRole } = setup();
    const displayImage = getByRole('img');
    expect(displayImage).toHaveAttribute('src', '/live-auction-card-cover.svg');
    expect(displayImage).toHaveAttribute('alt', '');
  });

  it('should test the Attribute section of the LiveAuctionCard component has a header ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('attribute-header')).toBeInTheDocument();
  });

  it('should test the Characters section of the LiveAuctionCard component has a header', () => {
    const { getByTestId } = setup();
    expect(getByTestId('characters-header')).toBeInTheDocument();
  });
  it('should test the LiveAuctionCard component has a GENRE_OPTIONS displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('GENRE_OPTIONS')).toBeInTheDocument();
  });

  it('should test the LiveAuctionCard component has a language type displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('language-type')).toBeInTheDocument();
  });

  it('should test the LiveAuctionCard component has an age rating displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('age-rating')).toBeInTheDocument();
  });

  it('should test the LiveAuctionCard component has characters displayed ', () => {
    const { getAllByTestId } = setup();
    const first_character = getAllByTestId('characters')[0];
    expect(first_character).toHaveTextContent(/spiderman/i);

    const second_character = getAllByTestId('characters')[1];
    expect(second_character).toHaveTextContent(/magnito/i);
  });
});
