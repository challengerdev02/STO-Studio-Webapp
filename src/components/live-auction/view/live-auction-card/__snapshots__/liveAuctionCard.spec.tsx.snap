// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveAuctionCard component should match with snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL hWhPCO"
    data-testid="live-auction-card"
  >
    <div
      class="sc-gsDKAQ bGfeRP"
    >
      <img
        alt=""
        class="sc-dkPtRN dnOmnC"
        data-testid="display-image"
        src="/live-auction-card-cover.svg"
      />
    </div>
    <div
      class="sc-hKwDye feTgcT"
    >
      <div
        class="sc-eCImPb doHkEG"
      >
        <h5
          class="sc-jRQBWg cEjJxq"
          data-testid="attribute-header"
        >
          Attributes
        </h5>
        <div
          class="sc-gKclnd bITziK"
        >
          <p
            class="sc-iCfMLu cHwwON"
          >
            <span
              class="sc-furwcr MfXef"
            >
              GENRE_OPTIONS
            </span>
            <span
              class="sc-furwcr MfXef"
              data-testid="GENRE_OPTIONS"
            >
              Science Fiction
            </span>
          </p>
          <p
            class="sc-iCfMLu cHwwON"
          >
            <span
              class="sc-furwcr MfXef"
            >
              Language
            </span>
            <span
              class="sc-furwcr MfXef"
              data-testid="language-type"
            >
              Japanese
            </span>
          </p>
          <p
            class="sc-iCfMLu cHwwON"
          >
            <span
              class="sc-furwcr MfXef"
            >
              Age Rating:
            </span>
            <span
              class="sc-furwcr MfXef"
              data-testid="age-rating"
            >
              13 and above
            </span>
          </p>
        </div>
      </div>
      <div
        class="sc-eCImPb doHkEG"
      >
        <h5
          class="sc-jRQBWg cEjJxq"
          data-testid="characters-header"
        >
          Characters
        </h5>
        <div
          class="sc-gKclnd bITziK"
        >
          <p
            class="sc-iCfMLu cHwwON"
          >
            <span
              class="sc-furwcr MfXef"
              data-testid="characters"
            >
              Spiderman
            </span>
          </p>
          <p
            class="sc-iCfMLu cHwwON"
          >
            <span
              class="sc-furwcr MfXef"
              data-testid="characters"
            >
              Magnito
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
